src/components/app/App.js

```diff
@@ -13,61 +13,149 @@ import './App.scss';
 import 'bootstrap/dist/css/bootstrap.min.css';
 
 
+
 // todo 1) вот наши опции сортировки, мы будем их использовать в DropDown (уже объявлены)
-const sortingOptions = ['Sort By Default', 'Sort By Author'];
+const sortingOptions = ['Sort By Default', 'Sort By Author', 'Sort By Popular'];
+const filteringOptions = ['Filter By Default', 'Filter With Img', 'Filter With Comments'];
 
 class App extends Component {
   state = {
     posts: [...postsList],
-    // todo: 2) добавить под ключом selectedOption значение sortingOptions[0] (она будет хранить выбранную в данный моменит опцию)
+    // ++todo: 2) добавить под ключом sortedOption значение sortingOptions[0] (она будет хранить выбранную в данный моменит опцию)
+    sortedOption: sortingOptions[0],
+    filteredOption: filteringOptions[0]
   };
+  // renderArr = [];
+
+  renderPosts = (selectedOption) => {
+    // this.renderArr = [...postsList];
+    let madeArr;
+    if (sortingOptions.includes(selectedOption)) {
+      madeArr = this.onSort(selectedOption, this.onFilter(this.state.filteredOption));
+    } else if (filteringOptions.includes(selectedOption)) {
+      madeArr = this.onSort(this.state.sortedOption, this.onFilter(selectedOption));
+    } else {return null}
 
-  renderList = () => {
-    const res = [];
+    this.setState({
+      posts: madeArr
+    })
+  }
```

Comment:
First of all, we should name functions starting with `render...` only if they return some jsx
if function only do some logic with data it shouldn't be named starting with `render...`

Reply:
The approach is correct

Reply:
As an improvement, I can suggest rewrite code a little like this:
````
    if (!sortingOptions.includes(selectedOption) || !filteringOptions.includes(selectedOption)) { return; }

const sortOption = sortingOptions.includes(selectedOption)  ? selectedOption : this.state.sortedOption;
const filterOption =  filteringOptions.includes(selectedOption) ? selectedOption : this.state.filteredOption;
      madeArr = this.onSort(sortOption, this.onFilter(filterOption));

.....
````